caracal/workers/selfcal_worker.py

```diff
@@ -360,6 +360,19 @@ def worker(pipeline, recipe, config):
             if not os.path.exists(os.path.join(pipeline.msdir, m)):
                 raise IOError(
                     "MS file {0:s}, to transfer gains to, does not exist. Please check that it is where it should be.".format(m))
+            # Check if a model subtraction has already been done
+            with table('{0:s}/{1:s}'.format(pipeline.msdir, m), readonly=False) as ms_table:
+                try:
+                    caracal.log.info(f"Checking if the model has been subtracted in {m}")
+                    nModelSub = ms_table.getcolkeyword('CORRECTED_DATA', 'modelSub')
+                    nModelSub = 0
+                    ms_table.putcolkeyword('CORRECTED_DATA', 'modelSub', nModelSub)
+                    nModelSub = ms_table.getcolkeyword('CORRECTED_DATA', 'modelSub')
+                    caracal.log.info(f"Model has been subtracted, reseting the counter to {nModelSub}")
+                    ms_table.close()     
+                except RuntimeError:
+                    caracal.log.info(f"No subtraction found in {m}")
+                    pass
```

Comment:
There is no need for a `pass` statement here.

Reply:
Ohhhh you're right. Because the pipeline will keep going with the `log.info ` .

Reply:
Exactly.